zipline/assets/futures.py

```diff
@@ -0,0 +1,128 @@
+from pandas import Timedelta
+
+from zipline.finance.trading import TradingEnvironment
+
+
+class FutureChain(object):
+    """ Allows users to look up future contracts.
+
+    Parameters
+    ----------
+    trading_algorithm : TradingAlgorithm
+    root_symbol : str, optional
+    as_of_date : pandas.Timestamp
+    """
+
+    def __init__(self, trading_algorithm, root_symbol='', as_of_date=None):
+        self.root_symbol = root_symbol.upper()
+
+        # If an as_of_date is provided, self._as_of_date uses that
+        # value, otherwise None. This attribute backs the as_of_date property.
+        if as_of_date:
+            self._as_of_date = (TradingEnvironment.instance().normalize_date(
+                                as_of_date))
+        else:
+            self._as_of_date = None
+
+        # Maintain a reference to the TradingAlgorithm which holds this
+        # FutureChain instance, so we can get its datetime.
+        self._trading_algorithm = trading_algorithm
```

Comment:
This may get real interesting with serialization. I don't know how we can improve this situation, but if there is one thing I'd like to change, it is how we hook-in to the algo's dt.